### PR TITLE
*: update dependencies

### PR DESCRIPTION
**Pull request by paulfantom**
Updating dependencies in preparation for 0.49.0 release

/cc @prometheus-operator/prometheus-operator-reviewers 

**Release Note Template (will be copied)**

```release-note:REPLACEME

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,38 +6,38 @@ require (
...
+	github.com/thanos-io/thanos v0.21.1
 	golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
 	google.golang.org/protobuf v1.26.0
 	gopkg.in/alecthomas/kingpin.v2 v2.2.6
 	gopkg.in/yaml.v2 v2.4.0
 	k8s.io/api v0.21.0
 	k8s.io/apiextensions-apiserver v0.21.0
-	k8s.io/apimachinery v0.21.0
+	k8s.io/apimachinery v0.21.2
 	k8s.io/client-go v12.0.0+incompatible
```
Comment (simonpasquier) [inline go.mod:36]: `k8s.io/client-go` in the replace section would need a bump?

File: go.mod [hunk 2]
```diff
@@ -6,38 +6,38 @@ require (
...
 	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.48.1
 	github.com/prometheus-operator/prometheus-operator/pkg/client v0.48.1
-	github.com/prometheus/alertmanager v0.21.1-0.20210422101724-8176f78a70e1
-	github.com/prometheus/client_golang v1.10.0
-	github.com/prometheus/common v0.23.0
-	github.com/prometheus/prometheus v1.8.2-0.20210518124745-db7f0bcec27b
+	github.com/prometheus/alertmanager v0.22.2
+	github.com/prometheus/client_golang v1.11.0
+	github.com/prometheus/common v0.29.0
+	github.com/prometheus/prometheus v1.8.2-0.20210519120135-d95b0972505f
```
Comment (simonpasquier) [inline go.mod:26]: can we make it the same commit ID as the latest supported version of Prometheus?

**Comment by paulfantom** [comment]
Additionally updated e2e CI job to test against k8s 1.21 and use better cluster bootstrap process (taken from https://github.com/prometheus-operator/kube-prometheus/pull/1237).

**Review by simonpasquier — COMMENTED** [review]
lgtm

**Pull request merged**